core/include/detray/propagator/rk_stepper.ipp

```diff
@@ -410,21 +410,13 @@ DETRAY_HOST_DEVICE auto detray::rk_stepper<
 
         point3_type dpos1 = pos;
         dpos1[i] += delta;
-        const auto bvec1_tmp =
+        const vector3_type bvec1 =
```

Comment:
I don't think a covfie field can be made to return an algebra plugins vector?
But maybe we can call the constructor instead of assigning everything 
```
        vector3_type bvec1{bvec1_tmp[0u], bvec1_tmp[1u], bvec1_tmp[2u]};
```